### ISSUE TITLE
segfault on macOS when closing window via titlebar button

### ISSUE THREAD

**Issue by cmyr** [xi-editor/druid#157]
That is, the red 'traffic light' button.

This is an invalid access, which suggests we're trying to send a message to an object that has been deallocated. My first hunch would be that the window is getting deallocated and is receiving a method from the view, or vice-versa. A first step to debug this would be to try the 'Zombies' instrument in Instruments.app.

**Comment by futurepaul** [comment]
Had some fun with Instruments.app. When I create a new window, Leak Check usually complains about roughly 3,000 leaks. A large of majority of the leaks (almost 3,000) were allocated prior to the creation of the second window.

On a hunch I added autorelease to content_view
```rust
let content_view = window.contentView();
content_view.autorelease();
```

Now when I run the multiwin example I can create windows and close them with the traffic light button without issue.

But!

When I record with Instruments.app I get a segfault:
```
2019-09-18 11:39:20.290 multiwin[31066:4741399] SetMenuTitleWithCFString(menuRef, (CFStringRef)[submenu title]) returned error -50 on line 3516 in -[NSCarbonMenuImpl _itemLostMagicAppleMenuStatus:]
2019-09-18 11:39:20.292 multiwin[31066:4741399] *** Assertion failure in -[NSView dealloc], /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1875.10.100/AppKit.subproj/NSView.m:1678
2019-09-18 11:39:20.294 multiwin[31066:4741399] *** -[NSAutoreleasePool drain]: This pool has already been drained, do not release it (double release).
```

(I don't get this same error message for the original segfault)

**Comment by cmyr** [comment]
We should only need to call `autorelease` on things that we explicitly create, and even then not always; for instance windows don't need `autorelease` by default, because they release themselves when they're closed (at least, this is my best understanding).

Anyway investigated this a bit; we were leaking tons of strings (and now menu items) and the stoplight crash was an over-release of window, I'm pretty sure; to the best of my understanding it's fixed in #166.
